pkg/console/operator/sync_v400.go

```diff
@@ -177,36 +212,44 @@ func SyncConsoleConfig(co *consoleOperator, consoleConfig *configv1.Console, rou
 	if err != nil {
 		return nil, err
 	}
-	logrus.Printf("Updating console.config.openshift.io with hostname: %v \n", host)
-	consoleConfig.Status.ConsoleURL = util.HTTPS(host)
+	httpsHost := util.HTTPS(host)
+	if consoleConfig.Status.ConsoleURL != httpsHost {
+		logrus.Printf("Updating console.config.openshift.io with hostname: %v \n", host)
+	}
+	consoleConfig.Status.ConsoleURL = httpsHost
 	return co.consoleConfigClient.UpdateStatus(consoleConfig)
 }
 
-func SyncDeployment(co *consoleOperator, recorder events.Recorder, operatorConfig *operatorv1.Console, cm *corev1.ConfigMap, serviceCAConfigMap *corev1.ConfigMap, sec *corev1.Secret) (*appsv1.Deployment, bool, error) {
-	logrus.Printf("validating console deployment...")
-	requiredDeployment := deploymentsub.DefaultDeployment(operatorConfig, cm, serviceCAConfigMap, sec)
+func SyncDeployment(co *consoleOperator, recorder events.Recorder, operatorConfig *operatorv1.Console, cm *corev1.ConfigMap, serviceCAConfigMap *corev1.ConfigMap, sec *corev1.Secret, rt *routev1.Route) (*appsv1.Deployment, bool, error) {
+	requiredDeployment := deploymentsub.DefaultDeployment(operatorConfig, cm, serviceCAConfigMap, sec, rt)
 	expectedGeneration := getDeploymentGeneration(co)
+	genChanged := operatorConfig.ObjectMeta.Generation != operatorConfig.Status.ObservedGeneration
+
+	if genChanged {
+		logrus.Printf("deployment generation changed from %s to %s \n", operatorConfig.ObjectMeta.Generation, operatorConfig.Status.ObservedGeneration)
+	}
+
+	deploymentsub.LogDeploymentAnnotationChanges(co.deploymentClient, requiredDeployment)
+
 	deployment, deploymentChanged, applyDepErr := resourceapply.ApplyDeployment(
 		co.deploymentClient,
 		recorder,
 		requiredDeployment,
 		expectedGeneration,
 		// redeploy on operatorConfig.spec changes
-		operatorConfig.ObjectMeta.Generation != operatorConfig.Status.ObservedGeneration,
+		genChanged,
 	)
 
 	if applyDepErr != nil {
 		logrus.Errorf("%q: %v \n", "deployment", applyDepErr)
 		return nil, false, applyDepErr
 	}
-	logrus.Println("deployment exists and is in the correct state")
 	return deployment, deploymentChanged, nil
 }
 
 // applies changes to the oauthclient
 // should not be called until route & secret dependencies are verified
 func SyncOAuthClient(co *consoleOperator, operatorConfig *operatorv1.Console, sec *corev1.Secret, rt *routev1.Route) (*oauthv1.OAuthClient, bool, error) {
-	logrus.Printf("validating oauthclient...")
```

Comment:
For the most part, these lines seem to be spam.  We should really only need a line to signal kicking the sync, not a ton of repetitive noise.